### PR TITLE
TST: use manylinux2010 docker instead of ubuntu

### PR DESCRIPTION
**Pull request by mattip**
xref MacPython/numpy-wheels#73

If we already are using a docker to test 32-bit linux, let's use the manylinux2010 one.

### CODE REVIEW

**Comment by mattip** [comment]
This is now consistent with what I see locally when I use the docker to test. The failure fo `sin` is due to very large numbers escaping the AVX functions, and glibc's implementation is not consistent across platforms.

**Review by charris — APPROVED** [review]

**Comment by charris** [comment]
LGTM, I'll let you decide when to put it in due to the ongoing test fix.

File: azure-pipelines.yml
Comment (charris) [inline azure-pipelines.yml]: 'multilinux' <- 'manylinux'

I've made that same mistake so many times the name should probably be changed :)

Comment (mattip) [reply azure-pipelines.yml]: fixed

**Comment by mattip** [comment]
rebased on top of gh-15598

**Pull request merged**
